core/ephemeral_key_provider_opkssh.go

```diff
@@ -0,0 +1,48 @@
+package core
+
+import (
+	"bytes"
+	"errors"
+	"os/exec"
+
+	"golang.org/x/crypto/ssh"
+)
+
+// OpksshEphemeralKeyProvider is a prototype implementation of EphemeralKeyProvider
+// that interacts with the opkssh CLI tool to obtain ephemeral SSH keys.
+type OpksshEphemeralKeyProvider struct {
+	// Path to the opkssh CLI executable
+	OpksshPath string
+}
+
+// GetEphemeralKey runs the opkssh CLI to obtain the ephemeral private key.
+func (p *OpksshEphemeralKeyProvider) GetEphemeralKey() ([]byte, error) {
+	if p.OpksshPath == "" {
+		p.OpksshPath = "opkssh"
+	}
+
+	cmd := exec.Command(p.OpksshPath, "key", "export", "--private")
```

Comment:
opkssh is implemented in Go; can't we use its Go API instead of invoking a cli command?